apps/oxlint/src/command/lint.rs

```diff
@@ -62,14 +66,49 @@ pub struct BasicOptions {
     /// Oxlint configuration file (experimental)
     ///  * only `.json` extension is supported
     ///  * tries to be compatible with the ESLint v8's format
+    ///
+    /// If not provided, Oxlint will look for the following files in the current
+    /// directory (in order):
+    ///
+    /// - `.oxlintrc.json`
+    /// - `oxlintrc.json`
+    /// - `.oxlint.json`
+    /// - `oxlint.json
     #[bpaf(long, short, argument("./oxlintrc.json"))]
-    pub config: Option<PathBuf>,
+    config: Option<PathBuf>,
 
     /// TypeScript `tsconfig.json` path for reading path alias and project references for import plugin
     #[bpaf(argument("./tsconfig.json"), hide_usage)]
     pub tsconfig: Option<PathBuf>,
 }
 
+impl BasicOptions {
+    const DEFAULT_OXLINTRC_NAMES: [&'static str; 4] =
+        [".oxlintrc.json", "oxlintrc.json", ".oxlint.json", "oxlint.json"];
```

Comment:
Maybe we should add ESLint config names as well. What do you think?

I also think `.oxlint.json` can be higher than `oxlintrc.json`, I have no heuristic to back this up but I feel most projects use hidden files for these configurations.

Reply:
I'm not doing the former based on Gilad's desire to have separate configs. He wants separation between oxlint/eslint configuration. Thoughts?

Reply:
Also, should we support `.oxlintrc`?

Reply:
> I'm not doing the former based on Gilad's desire to have separate configs. He wants separation between oxlint/eslint configuration. Thoughts?

Good point; I'm not sure about this, Maybe we can have it at the very end of our check so we only use it if there is no oxlint specific file. I don't know exactly what is our plan to satisfy their configuration requirements, That's why I also think it can wait until we sort the fundamentals out.

> Also, should we support .oxlintrc?

Nice catch, The `.eslint` file is common in the wild so that seems like a nice addition to avoid breaking people's habits. It's bad UX if someone does this which is normal in eslint land and doesn't get the expected behavior or - if we don't want to allow non-json configs as our default files - get a warning.